CONTRIBUTING.md

```diff
@@ -0,0 +1,219 @@
+
+1. Create a new branch
+2. (Optional) Push it to github to share and for backup
+3. Make changes, commits
+4. Submit a pull request into **next** using github's [Pull Requests][PRs]
+   system
+
+### Creating a feature branch ###
+
+First get a copy of the [BOUT-dev] repository (or git pull to update an existing
+copy):
+
+    git clone git@github.com:boutproject/BOUT-dev.git
+    cd BOUT-dev
+
+Create a new branch **myfeature**, branching from **next**. Choose a descriptive
+name for **myfeature**, anything except "master" or "next".
+
+    git checkout next
+    git pull
+    git checkout -b myfeature     # Switched to a new branch "myfeature"
+
+### Pushing to github ###
+
+If you want to push your branch to BOUT-dev to share with other developers, run:
+
+    git push -u origin myfeature
+
+This command pushes **myfeature** to the central BOUT-dev repository (origin),
+and the -u flag adds it as a remote tracking branch.  After setting up the
+tracking branch, you can call "git push" without any parameters to push updates
+to **myfeature**.
+
+If another developer wants to try out this branch, they should clone the
+BOUT-dev repository (or update an existing one), then run
+
+    git checkout -b myfeature origin/myfeature
+
+*Note* If you do not have write access to the BOUT-dev repository, you can
+develop in your own fork and then send a pull request to someone with write
+access.
+
+### Making changes, commits ###
+
+Now you would make changes, commit changes and push as usual:
+
+    ... make changes ...
+    git add <files>
+    git commit
+    git push   # Pushes to origin/myfeature
+
+You can switch between branches using *checkout*:
+
+    git checkout master    # Switch to "master"
+    git checkout myfeature # Switch to "myfeature"
+
+### Merging into **next** ###
+
+Once your feature is complete, ask other developers to have a look by creating
+a [Pull Request][PRs] on the [BOUT-dev] page. One of the maintainers will review
+your code and merge it into **next**. They may give you comments to improve the
+code. You can make additional changes and push them to the same feature branch
+and they will be automatically added to the pull request.
+
+
+## Coding Style ##
+
+Code is read an order of magnitude more times than it is written. It's also
+written for *people* and not for the computer! For these reasons, it's important
+that we stick to some form of coding standards. The following coding style
+guidelines broadly follow
+the [LLVM Coding Standards][LLVM].
+
+These guidelines are intended to make code easier to read and therefore easier
+to understand. Being consistent in coding style also helps comprehension by
+reducing cognitive load.
+
+
+### Comments ###
+
+
+Comments in the code are vital to helping understanding. Comments that
+are embedded in the code should explain **why** something is done,
+rather than *how*. For documenting what functions and classes do, we
```

Comment:
Maybe a line break to clarify that doxygen is completely independent / opposite of what has been said before?
Further mention doxygen only in header files?